wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,435 @@
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class ItemHoverEvent {
+    /** The item stack to apply the event to. */
+    public var itemStack: ItemStack? = null
+    /** The item to apply the event to. */
+    public var item: Item? = null
+    /** The NBT of the item to apply the event to. */
+    public var nbt: NbtCompound? = null
+
+    /**
+     * Gets the item based off the [itemStack], [item] or [nbt] variables.
+     *
+     * @author NoComment1105
+     */
+    private fun getItem(): ItemStack {
+        return if (itemStack != null) {
+            itemStack!!
+        } else if (item != null) {
+            ItemStack(item).let {
+                it.nbt = nbt
+                it
+            }
+        } else if (nbt != null) {
+            ItemStack.fromNbt(nbt)
+        } else {
+            ItemStack.EMPTY
+        }
+    }
+
+    /**
+     * Creates the hover event.
+     *
+     * @author NoComment1105
+     */
+    public fun create(): HoverEvent {
+        return HoverEvent(
+            HoverEvent.Action.SHOW_ITEM,
+            HoverEvent.ItemStackContent(
+                itemStack ?: getItem()
+            )
+        )
+    }
+}
+
+/**
+ * The builder to create a hover event for an entity.
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class EntityHoverEvent {
+    /** The entity type to apply the hover event to. */
+    public var entityType: EntityType<Entity>? = null
+    /** The UUID of hte entity to apply the event to. */
+    public var uuid: UUID? = null
+    /** The name of the entity to apply the event to. */
+    public var name: Text? = null
+
+    /**
+     * Creates a hover event for an entity based off of [entityType], [uuid] and [name].
+     *
+     * @author NoComment1105
+     */
+    public fun create(): HoverEvent {
+        return HoverEvent(
+            HoverEvent.Action.SHOW_ENTITY,
+            HoverEvent.EntityContent(
+                entityType ?: EntityType.PLAYER,
+                uuid ?: UUID.randomUUID(),
+                name
+            )
+        )
+    }
+}
+
+/**
+ * The builder to create a hover event for text.
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class TextHoverEvent {
+    /** The text to apply the hover event to. */
+    public var text: Text? = null
+
+    /**
+     * Creates a hover event for the [text] provided.
+     *
+     * @author NoComment1105
+     */
+    public fun create(): HoverEvent {
+        return HoverEvent(
+            HoverEvent.Action.SHOW_TEXT,
+            text ?: buildText {
+                empty()
+            }
+        )
+    }
+}
+
+/**
+ * The builder to create a [ClickEvent].
+ *
+ * @author NoComment1105
+ */
+public class QklClickEvent {
+    /** The action to perform on the click. */
+    private var action: ClickEvent.Action? = null
+    /** The value for the event. */
+    private var value: String? = null
+
+    /** Open's a URL. */
+    public var openUrl: String = ""
+        set(url) = run {
```

Comment:
Why are you wrapping in ` = run`? Setters are full functions and can be treated the same way.
```suggestion
        set(url) {
```